torch/jit/annotations.py

```diff
@@ -105,15 +105,52 @@ def parse_type_line(type_line):
 
 def get_type_line(source):
     """Tries to find the line containing a comment with the type annotation."""
-    lines = source.split('\n')
+    type_comment = '# type:'
 
-    type_line = None
-    for line in lines:
-        if '# type:' in line:
-            type_line = line.strip()
-            break
+    lines = source.split('\n')
+    lines = [(line_no, line) for line_no, line in enumerate(lines)]
+    lines = list(filter(lambda line: type_comment in line[1], lines))
 
-    return type_line
+    if len(lines) == 0:
+        return None
+    elif len(lines) == 1:
+        # Only 1 type line, quit now
+        return lines[0][1].strip()
```

Comment:
I think this might have the problem Adam mentioned::

`type: anywhere inside the body of the function, which is not great. You can e.g. have nested functions, and the outer one will see the inner annotation!`

Reply:
For a single type line we had that issue previously and the only way around that I can figure is using the indentation to indicate if it's relevant or not. For multiline, the code after this makes sure they're all contiguous.

Reply:
I'm not sure this problem is really resolved. Something like the following would fail with a weird error message about having three function parameters with only one annotated, right ? 
```
@torch.jit.script
def fn(a,
       b,
       c  # type: int
       ):
```

Where as in python 3 the following would work

```
@torch.jit.script
def fn(a, b, c: int ):
```

Reply:
I think we should just land without that feature (and recommend people use Python 3 if they want it) and treat this as just sugar around Python 2 style type comments. Adding Tensor default args is difficult without changing a lot of the infrastructure around getting types or re-parsing all the args in Python.